source/Halibut/IHalibutRuntime.cs

```diff
@@ -15,7 +15,7 @@ public enum UnauthorizedClientConnectResponse
         TrustAndAllowConnection
     }
 
-    public interface IHalibutRuntime : IDisposable
+    public interface IHalibutRuntime : IDisposable, IAsyncDisposable
```

Comment:
Are we missing some test coverage/some of or tests using the DisposeAsync method?

E.g. `LatestClientAndLatestServiceBuilder.ClientAndService` calls Dispose on the HalibutRuntimes. We could cover both paths by making it Dispose when we are sync and DisposeAsync when we are forcing it to be async

Reply:
Good thinking. Done!